clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -440,7 +457,9 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                           TRACE_MARKER_TYPE_KERNEL_XFER)) ||
                     // We expect a gap on a window transition.
                     shard->window_transition_ ||
-                    shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER,
+                    shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER ||
+                    saw_repeated_syscall_instrs_with_same_pc ||
+                    shard->prev_instr_was_syscall_,
```

Comment:
I'm confused about this: if the previous instr is a syscall we don't report any discontinuities?

Reply:
Ack. Removed it: After changes with the decoding logic, it is not needed. Previously it was reporting discontinuities at the syscall instr before we detected the double syscall on the following instr.